### PR TITLE
Allow pure evaluation with md5 checksums

### PR DESCRIPTION
**Pull request by balsoft**
Use `hash` argument to FODs to allow for md5 hashes.

### CODE REVIEW

**Comment by vapourismo** [comment]
Really nice! Thanks for the contribution.

With which package did you test this?

File: nix/packages/ocaml/opam2nix.nix [hunk 1]
```diff
@@ -24,6 +26,8 @@ buildDunePackage {
     ppx_deriving
     cmdliner
     zarith
+    hex
+    base64
```
Comment (vapourismo) [inline nix/packages/ocaml/opam2nix.nix:30]: The other `opam*2nix` packages in this directory also need this change.

Comment (vapourismo) [reply nix/packages/ocaml/opam2nix.nix:30]: The Nix examples in `examples/*.nix` also most likely need the additional dependencies as they test building this project using itself.

**Comment by balsoft** [comment]
> With which package did you test this?

Removed the `sha256` from `hacl-star`, and built it with `makePackageSet`

**Review by vapourismo — APPROVED** [review]
Looks great, thanks!

**Pull request merged**